atlasexec/atlas.go

```diff
@@ -300,7 +306,28 @@ func (c *Client) MigrateLint(ctx context.Context, params *MigrateLintParams) (*S
 		args = append(args, "--latest", strconv.FormatUint(params.Latest, 10))
 	}
 	args = append(args, params.Vars.AsArgs()...)
-	return jsonDecode[SummaryReport](c.runCommand(ctx, args, validJSON))
+	return args
+}
+
+// MigrateLint runs the 'migrate lint' command.
+func (c *Client) MigrateLint(ctx context.Context, params *MigrateLintParams) (*SummaryReport, error) {
+	if params.Writer != nil || params.Web {
+		return nil, errors.New("custom Writer or Web reporting are not supported with MigrateLint, use MigrateLintError")
+	}
+	args := lintArgs(params)
+	reader, err := c.runCommand(ctx, args, validJSON)
+	return jsonDecode[SummaryReport](reader, err)
+}
+
+// MigrateLintError runs the 'migrate lint' command, the output is written to params.Writer
+func (c *Client) MigrateLintError(ctx context.Context, params *MigrateLintParams) error {
+	args := lintArgs(params)
+	reader, err := c.runCommand(ctx, args)
```

Comment:
```suggestion
	r, err := c.runCommand(ctx, args)
```
Use r for readers. it's common enough